src/fluree/db/ledger/bootstrap.clj

```diff
@@ -160,74 +163,90 @@
-                                     (flake/new-flake t (get pred->id "_tx/nonce") timestamp t true)
-                                     (flake/new-flake block-t (get pred->id "_block/number") 1 block-t true)
-                                     (flake/new-flake block-t (get pred->id "_block/instant") timestamp block-t true)
-                                     (flake/new-flake block-t (get pred->id "_block/transactions") -1 block-t true)
-                                     (flake/new-flake block-t (get pred->id "_block/transactions") -2 block-t true)))
-          hash             (get-block-hash flakes*)
-          block-flakes     [(flake/new-flake block-t (get pred->id "_block/hash") hash block-t true)
-                            (flake/new-flake block-t (get pred->id "_block/ledgers") auth-subid block-t true)]
-          flakes+block     (into flakes* block-flakes)
-
-          {:keys [spot psot post opst]} novelty
-          new-db           (assoc db :block block
-                                     :t block-t
-                                     :ecount genesis-ecount
-                                     :novelty (assoc novelty :spot (into spot flakes+block)
-                                                             :psot (into psot flakes+block)
-                                                             :post (into post (filter #(index-pred (.-p ^Flake %)) flakes+block))
-                                                             :opst (into opst (filter #(ref-pred (.-p ^Flake %)) flakes+block))
-                                                             :size (flake/size-bytes flakes+block))
-                                     :stats (assoc stats :flakes (count flakes+block)
-                                                         :size (flake/size-bytes flakes+block)))
-
-          block-data       {:block  block
-                            :t      block-t
-                            :flakes flakes+block
-                            :hash   hash
-                            :txns   {txid {:t   t
-                                           :cmd cmd
-                                           :sig sig}}}
-          _                (<? (storage/write-block conn network dbid block-data))
+          _             (when (or (txproto/ledger-exists? group network dbid)
+                                  ;; also check for block 1 on disk as a precaution
+                                  (<? (storage/block conn network dbid 1)))
+                          (throw (ex-info (str "Ledger " network "/$" dbid " already exists! Create unsuccessful.")
+                                          {:status 500
+                                           :error  :db/unexpected-error})))
+          master-authid (crypto/account-id-from-message cmd sig)
+          block         (boostrap-memory-db conn [network dbid] {:master-auth-id master-authid :txid txid :cmd cmd :sig sig})
```

Comment:
Ah, I see that `bootstrap-memory-db` was itself an extract-to-separate-fn refactor of sorts. :)

Reply:
Yes, and was trying to change minimum amount possible here - goal was not a refactor job, but a ticket could be created to refactor if you wanted.